zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs

```diff
@@ -130,38 +192,38 @@ impl InputSource for TransactionRecordsById {
             .map(|(id, value)| NonNegativeAmount::from_u64(value).map(|value| (id, value)))
             .collect::<Result<Vec<_>, _>>()
             .map_err(InputSourceError::InvalidValue)?;
-        unselected.sort_by_key(|(_id, value)| *value);
+        unselected.sort_by_key(|(_id, value)| *value); // from smallest to largest
+        let dust_spendable_index =
+            unselected.partition_point(|(_id, value)| *value <= MARGINAL_FEE);
+        let dust_notes: Vec<_> = unselected.drain(..dust_spendable_index).collect();
         let mut selected = vec![];
         let mut index_of_unselected = 0;
 
         loop {
-            // if no unselected notes are available, return the currently selected notes even if the target value has not been reached
-            if unselected.is_empty() {
-                break;
-            }
```

Comment:
why move this below all the other processing with updating target value? if its at the top it can just break and avoid unecessary calculation?

Reply:
It's not clear to me which order is better.